File: client/src/app/modules/quotation/services/act-type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ActType } from '../../quotation/model/ActType';

@Injectable({
  providedIn: 'root'
})
export class ActTypeService extends AppRestService<ActType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getActTypes() {
    return this.getListCached(new HttpParams(), "act-types");
  }

  addOrUpdateActType(actType: ActType) {
    this.clearListCache(new HttpParams(), "act-types");
    return this.addOrUpdate(new HttpParams(), "act-type", actType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/ActTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.ActType;

public interface ActTypeService {
    public List<ActType> getActTypes();

    public ActType getActType(Integer id);

    public ActType addOrUpdateActType(ActType actType);
}
